### PR TITLE
add optional clip filtering by tags

### PR DESCRIPTION
**Pull request by salma-remyx**
To help users further refine a dataset, we introduce an optional content based filtering step which includes/excludes image samples based on tags of interest.

### CODE REVIEW

File: config/config.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# config.yaml
+directories:
+  image_dir: /path/to/images
```
Comment (smellslikeml) [inline config/config.yaml:3]: update README instructions for this config.yaml

File: config/config.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+  output_dir: /path/to/output
+
+arguments:
+  include_tags: ""
```
Comment (smellslikeml) [inline config/config.yaml:7]: put in the tags we use

File: docker/depth_processor/process_depth.py [hunk 3]
```diff
@@ -6,60 +6,45 @@
...
 
-        # Convert records to a pandas DataFrame
-        df = pd.DataFrame(records)
 
-        # Save the DataFrame to a .pkl file
-        output_filepath = os.path.join(output_dir, f"chunk_{chunk_index}.pkl")
-        df.to_pickle(output_filepath)
+def main(output_dir):
+    for filename in os.listdir(output_dir):
+        if filename.endswith(".pkl"):
```
Comment (smellslikeml) [inline docker/depth_processor/process_depth.py:27]: update dataloader with HF in next PR

File: vqasynth/datasets/utils.py [hunk 4]
```diff
@@ -1,11 +1,46 @@
...
 import matplotlib
 import matplotlib.cm
+from torchvision.datasets import CIFAR100
+
+class ImageTagger:
+    def __init__(self, model_name='ViT-B/32', device=None):
+        """Initialize the CLIP model and load CIFAR100 dataset."""
+        self.device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+        self.model, self.preprocess = clip.load(model_name, self.device)
+        self.cifar100 = CIFAR100(root=os.path.expanduser("~/.cache"), download=True, train=False)
```
Comment (smellslikeml) [inline vqasynth/datasets/utils.py:18]: what is this doing here?

File: vqasynth/datasets/utils.py [hunk 5]
```diff
@@ -1,11 +1,46 @@
...
+import os
+import clip
 import base64
 import numpy as np
 import torch
 from PIL import Image
 
 import matplotlib
 import matplotlib.cm
+from torchvision.datasets import CIFAR100
```
Comment (smellslikeml) [inline vqasynth/datasets/utils.py:11]: looks like you've copied CLIP's github demo into here

File: vqasynth/datasets/utils.py [hunk 6]
```diff
@@ -1,11 +1,46 @@
...
+        with torch.no_grad():
+            image_features = self.model.encode_image(image_input)
+            text_features = self.model.encode_text(self.text_inputs)
+
+        image_features /= image_features.norm(dim=-1, keepdim=True)
+        text_features /= text_features.norm(dim=-1, keepdim=True)
+        similarity = (100.0 * image_features @ text_features.T).softmax(dim=-1)
+        _, indices = similarity[0].topk(top_k)
+
+        return [self.cifar100.classes[index] for index in indices]
```
Comment (smellslikeml) [inline vqasynth/datasets/utils.py:43]: you should be saving the embeddings to another field in the dataset
we'll want to extract embeddings once and apply tagging using different sets of labels over multiple stages

File: config/config.yaml (hunk 2 above)
Comment (smellslikeml) [reply config/config.yaml:7]: Let's update the ImageTagger, decoupling the work to create/save CLIP embeddings from the work to match against tags.

Tags will be high-level, thematic descriptors to help filter out unsuitable images like screenshots/art/aerial imagery and to select for the typical first person perspectives of indoor/outdoor scenes. Ideal scenes will have a few well-separated objects, easily distinguished by appearance or spatial relation, convex objects occupying large parts of the image plane.

Generating embeddings in one pass for all the images, we can use zero-shot matching with text embeddings to filter over additional stages. For example, tag images by characteristic scale length "1 inch scale", "1 foot scale", "10 foot scale", "100 foot scale"
